CRM/Sepa/Logic/Batching.php

```diff
@@ -482,6 +480,57 @@ static function closeEnded() {
    **                                                                        **
    ****************************************************************************/
 
+  public static function getOrCreateTransactionGroup(
+    int $creditor_id,
+    string $mode,
+    string $collection_date,
+    ?int $financial_type_id,
+    int $notice,
+    array &$existing_groups
+  ): int {
+    $group_status_id_open = (int) CRM_Core_PseudoConstant::getKey('CRM_Batch_BAO_Batch', 'status_id', 'Open');
+
+    if (!isset($existing_groups[$collection_date][$financial_type_id ?? 0])) {
+      // this group does not yet exist -> create
+
+      // find unused reference
+      $reference = self::getTransactionGroupReference($creditor_id, $mode, $collection_date, $financial_type_id);
+
+      $group = civicrm_api('SepaTransactionGroup', 'create', array(
+        'version'                 => 3,
+        'reference'               => $reference,
+        'type'                    => $mode,
+        'collection_date'         => $collection_date,
+        // Financial type may be NULL if not grouping by financial type.
+        'financial_type_id'       => $financial_type_id,
+        'latest_submission_date'  => date('Y-m-d', strtotime("-$notice days", strtotime($collection_date))),
+        'created_date'            => date('Y-m-d'),
+        'status_id'               => $group_status_id_open,
+        'sdd_creditor_id'         => $creditor_id,
+      ));
+      if (!empty($group['is_error'])) {
+        // TODO: Error handling
+        Civi::log()->debug("org.project60.sepa: batching:syncGroups/createGroup ".$group['error_message']);
```

Comment:
I think this should be logged as error.

Reply:
Actually this should throw an exception, as code after this accesses `$groups`. There is just no error handling and has not been, which is bad and should be taken care of in a follow-up, though I've never seen this erroring in the wild.